.teamcity/settings.kts

```diff
@@ -226,12 +227,22 @@ project {
                 password("env.PGP_SIGNING_KEY_PASSPHRASE", "%pgpSigningPassphrase%")
             }
             steps {
+                script {
+                    scriptContent = """
+                        git config credential.helper '!f() { sleep 1; echo "username=${'$'}{GIT_USERNAME}"; echo "password=${'$'}{GIT_ACCESS_TOKEN}"; }; f' 
```

Comment:
This is a custom credential helper. The `!` is a special syntax from Git and means that it should be executed in a Shell.

Reply:
What is the purpose of `sleep 1`? :)

Reply:
I saw this solution [here](https://stackoverflow.com/questions/8536732/can-i-hold-git-credentials-in-environment-variables), where they claimed that on some systems this resolves a race condition. Nobody seems to know, though, who is actually racing. :)

I think it has to do with the fact that stdin is typically drained before a password prompt. That's why you cannot do things like `echo my_password | execute_something_which_is_requesting_a_password`. But I'm just guessing here.